packages/react-core/src/Authenticator/hooks/useAuthenticatorInitMachine/__tests__/useAuthenticatorInitMachine.spec.tsx

```diff
@@ -0,0 +1,47 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { useAuthenticatorInitMachine } from '..';
+import { useAuthenticator, UseAuthenticator } from '../../useAuthenticator';
+
+jest.mock('../../useAuthenticator');
+
+const initializeMachine = jest.fn();
+
+describe('useAuthenticatorInitMachine', () => {
+  beforeEach(() => {
+    initializeMachine.mockClear();
+  });
+
+  it('calls initializeMachine only once, even after subsequent rerenders', () => {
+    const route = 'setup';
+    let data = {};
+
+    (useAuthenticator as jest.Mock).mockReturnValue({
+      initializeMachine,
+      route,
+    } as unknown as UseAuthenticator);
+    const { rerender } = renderHook(() => useAuthenticatorInitMachine(data));
+
+    expect(initializeMachine).toHaveBeenCalledTimes(1);
+
+    // change the input props of the hook to get the useEffect to run again
+    data = { mutated: 'dataObject' };
```

Comment:
Prefer to avoid mutating test variables to achieve a desired behavior as it increases fragility and the behavior itself can be difficult for future maintainers to comprehend the context of. 

As an alternative, the `rerender` API probably provides handling it through the testing library, maybe try the [docs](https://react-hooks-testing-library.com/reference/api#rerender)

Reply:
Good call. I found a way to use the `rerender` API to handle this